src/table/interfaces.tsx

```diff
@@ -321,25 +318,20 @@ export namespace TableProps {
     validation?: (item: T, value: Optional<V>) => Optional<string>;
   }
 
-  export type ColumnEditConfig<ItemType, ValueType> =
-    | {
-        editConfig?: undefined;
-        cell: CellRenderer<ItemType>;
-      }
-    | {
-        editConfig: EditConfig<ItemType, ValueType>;
-        cell: EditableCellRenderer<ItemType, ValueType>;
-      };
-
-  export type ColumnDefinition<ItemType, ValueType = any> = {
+  export interface ColumnEditConfig<ItemType> {
+    editConfig?: EditConfig<ItemType>;
+    cell(item: ItemType, context?: CellContext<any>): React.ReactNode;
```

Comment:
I do not like that now there is optional chaining like `context?.isEditing` all over the place. Should we provide an empty context object for all cells?

Reply:
We do provide a readOnly context value even for non-editable cells. But the problem with requiring this value is packages downstream might not be providing this value. I guess the question is, Do we want to break those implementations?

Reply:
Since this is a parameter (i.e. a value _we_ provide rather than a value _consumers_ provide), I think it won't cause errors for consumers who don't use the parameter.

See https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgJKQLYBUCeAHFAbwF8BYAKFElkRQGEIAbRuge3AgA8wAeLAPhIUKYfPSaMAYgFcQCMMHYB1YGAAWAeTwL2cRsgC8yABSqIGAFxpMuAgBpkCdpG4B+Kw2ZsO3HnBA4-ACUhvzIAG6swAAmANwiYsieUrLyiiAq6gBKEACO0sBQENGGJmaW1ua2EA5OPmAeEt4uvP6BIQZhkTHx5BR1AM5gyOwQjcwycjoZqpra6XqlppgdYbLREDCgxb2Dw2AA7qzjKVPpmWo5+YXFS+WryOub23FAA

Reply:
My comment wasn't clear, sorry about that. What I meant to say is that some of our consumers intercept this call like this; this then breaks the type signature.

<img width="509" alt="image" src="https://user-images.githubusercontent.com/7447677/210774770-f99d7649-0988-42bd-92b9-237a3e134c6a.png">

Reply:
Namely, this code: https://github.com/awslabs/iot-app-kit/blob/main/packages/related-table/src/RelatedTable/RelatedTable.tsx#L22

Discussed in DM, this use is not matching our API contract. `cell` is only supposed to be called by us, and not anyone outside. As a quick fix, we will change their code to `columnToExpand?.cell(node, ...rest)`